### PR TITLE
Add combine data window

### LINKED ISSUE #86: Integrate processing functionality from vs
As much as I like terminal-based processing of the data, we should make things a bit more user friendly by bringing it into the main app so people without terminal access/knowledge can handle scouting data.

### PR DESCRIPTION
**Pull request by AndrewLester**
Resolves #86
Lots of style issues in here. I'll need suggestions because I literally can not use css. Aside from that, most of the js has worked in testing. (I have not done enough for this PR to be accepted yet)

- [x] Test javascript code with malformed json files and improper json files.
- [x] Change place csv files are saved to. (Another localStorage section?)

### CODE REVIEW

**Comment by ErikBoesen** [comment]
Congrats on PR #:100:

**Comment by ErikBoesen** [comment]
My recommendation on this would be to incorporate ALL the data processing into the View Data section. Also, yeah, don't worry about making the button red.

File: data.js [hunk 1]
```diff
@@ -3,13 +3,21 @@ const fs = require('fs');
 // Define <thead>, <tbody>, and warning vars to be filled later on.
 var thead = document.getElementsByTagName('thead')[0],
     tbody = document.getElementsByTagName('tbody')[0],
-    warning = document.getElementById('warning');
+    warning = document.getElementById('warning'),
+    combineFilesFlex = document.getElementById('combine-files'),
```
Comment (ErikBoesen) [inline data.js]: Sorta weird name

File: data.js [hunk 2]
```diff
@@ -3,13 +3,21 @@ const fs = require('fs');
...
+    fileInput = document.getElementById('file-input'),
+    fileList = document.getElementById('file-list'),
+    outputButton = document.getElementById('csv-button'),
+    fileName = document.getElementById('output-file');
```
Comment (ErikBoesen) [inline data.js]: `outputFileName`?

File: data.js [hunk 3]
```diff
@@ -3,13 +3,21 @@ const fs = require('fs');
 // Define <thead>, <tbody>, and warning vars to be filled later on.
 var thead = document.getElementsByTagName('thead')[0],
     tbody = document.getElementsByTagName('tbody')[0],
-    warning = document.getElementById('warning');
+    warning = document.getElementById('warning'),
+    combineFilesFlex = document.getElementById('combine-files'),
+    fileInput = document.getElementById('file-input'),
+    fileList = document.getElementById('file-list'),
```
Comment (ErikBoesen) [inline data.js]: These two names are kind of ambiguous as well.

**Comment by AndrewLester** [comment]
Tasks complete.

File: script.js [hunk 4]
```diff
@@ -21,7 +21,10 @@ var pg = {
 // Get date for file naming.
 var d = new Date();
 var home = process.env[(process.platform == 'win32') ? 'USERPROFILE' : 'HOME'];
-var path = home + (fs.existsSync(home + '/Desktop') ? '/Desktop' : '') + '/scoutdata_' + user.username + '_' + ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'][d.getMonth()] + '_' + d.getDate() + '_' + d.getFullYear() + '.json';
+var desktopPath = home + (fs.existsSync(home + '/Desktop') ? '/Desktop' : '');
+var path = desktopPath + '/scoutdata_' + user.username + '_' + ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'][d.getMonth()] + '_' + d.getDate() + '_' + d.getFullYear() + '.json';
+
+localStorage.desktopPath = desktopPath;
```
Comment (AndrewLester) [inline script.js]: I'll be using this for the USB sending button as well.

Comment (ErikBoesen) [reply script.js]: Just FYI, we should probably find a nicer way to send this data. This is totally fine for now though.

**Comment by ErikBoesen** [comment]
This is satisfactory. There are a couple things I'd like to change though, I will open them issues of their own in a minute.

**Pull request merged**